// @mocks-server/admin-api-client/dist-tsc/methods.js
import { BaseAdminApiClient } from "./entities";
export class AdminApiClient {
    constructor() {
        Object.defineProperty(this, "_adminApiClient", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        this._adminApiClient = new BaseAdminApiClient();
    }
    readAbout() {
        return this._adminApiClient.about.read();
    }
    readConfig() {
        return this._adminApiClient.config.read();
    }
    updateConfig(newConfig) {
        return this._adminApiClient.config.update(newConfig);
    }
    readAlerts() {
        return this._adminApiClient.alerts.read();
    }
    readAlert(id) {
        return this._adminApiClient.alert(id).read();
    }
    readCollections() {
        return this._adminApiClient.collections.read();
    }
    readCollection(id) {
        return this._adminApiClient.collection(id).read();
    }
    readRoutes() {
        return this._adminApiClient.routes.read();
    }
    readRoute(id) {
        return this._adminApiClient.route(id).read();
    }
    readVariants() {
        return this._adminApiClient.variants.read();
    }
    readVariant(id) {
        return this._adminApiClient.variant(id).read();
    }
    readCustomRouteVariants() {
        return this._adminApiClient.customRouteVariants.read();
    }
    useRouteVariant(id) {
        return this._adminApiClient.customRouteVariants.create({
            id,
        });
    }
    restoreRouteVariants() {
        return this._adminApiClient.customRouteVariants.delete();
    }
    configClient(config) {
        return this._adminApiClient.configClient(config);
    }
}
