// @mocks-server/cypress-commands/dist/helpers.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.isUndefined = exports.isTruthy = exports.isFalsy = exports.valueIsOneOf = exports.TRUTHY_VALUES = exports.FALSY_VALUES = exports.ADMIN_API_HTTPS_ENVIRONMENT_VAR = exports.ADMIN_API_PORT_ENVIRONMENT_VAR = exports.ADMIN_API_HOST_ENVIRONMENT_VAR = exports.ENABLED_ENVIRONMENT_VAR = exports.LOG_ENVIRONMENT_VAR = void 0;
exports.LOG_ENVIRONMENT_VAR = "MOCKS_SERVER_LOGS";
exports.ENABLED_ENVIRONMENT_VAR = "MOCKS_SERVER_ENABLED";
exports.ADMIN_API_HOST_ENVIRONMENT_VAR = "MOCKS_SERVER_ADMIN_API_HOST";
exports.ADMIN_API_PORT_ENVIRONMENT_VAR = "MOCKS_SERVER_ADMIN_API_PORT";
exports.ADMIN_API_HTTPS_ENVIRONMENT_VAR = "MOCKS_SERVER_ADMIN_API_HTTPS";
exports.FALSY_VALUES = [false, "false", 0, "0"];
exports.TRUTHY_VALUES = [true, "true", 1, "1"];
function valueIsOneOf(value, arrayOfValues) {
    return arrayOfValues.includes(value);
}
exports.valueIsOneOf = valueIsOneOf;
function isFalsy(value) {
    return valueIsOneOf(value, exports.FALSY_VALUES);
}
exports.isFalsy = isFalsy;
function isTruthy(value) {
    return valueIsOneOf(value, exports.TRUTHY_VALUES);
}
exports.isTruthy = isTruthy;
// eslint-disable-next-line @typescript-eslint/no-explicit-any
function isUndefined(value) {
    return typeof value === "undefined";
}
exports.isUndefined = isUndefined;
